src/librustc_typeck/collect.rs

```diff
@@ -150,12 +150,24 @@ crate fn placeholder_type_error(
     let mut sugg: Vec<_> =
         placeholder_types.iter().map(|sp| (*sp, type_name.to_string())).collect();
     if generics.is_empty() {
-        sugg.push((ident_span.shrink_to_hi(), format!("<{}>", type_name)));
+        sugg.push((span, format!("<{}>", type_name)));
     } else {
-        sugg.push((
-            generics.iter().last().unwrap().span.shrink_to_hi(),
-            format!(", {}", type_name),
-        ));
+        if let Some(arg) = generics.iter().find(|arg| match arg.name {
```

Comment:
`else if let ...`?

Reply:
I feel that `if let`/`else` can be harder to read for cases like these:

```
        if let Some(arg) = generics.iter().find(|arg| match arg.name {
            hir::ParamName::Plain(Ident { name: kw::Underscore, .. }) => true,
            _ => false,
        }) {
```

vs

```
        if let Some(arg) = generics.iter().find(|arg| {
            if let hir::ParamName::Plain(Ident { name: kw::Underscore, .. }) = arg.name {
                true
            } else {
                false
            }
        }) {
```

Reply:
ah, I see what you meant